### ISSUE TITLE
Resume Checkpoint - LR wrong?

### ISSUE THREAD

**Issue by CognitiveDiffusion** [tdrussell/diffusion-pipe#278]
Hey there tdrussel,

Pipe-Diffusion is still my go-to trainer for Wan and I love it. Thanks for all the work you put into this project.

I'm running a resume training run right now and I'm confused. I set the lr to 2e-5 (and I checked shell and it gets noted accordingly before training starts), but during training pipe shows lr=[0.0001]  for every step - which is 1e-4 afaik.

So either pipe shows the wrong lr or (that would be worse) trains at the wrong lr.

Maybe I'm reading the log wrong, but I felt like this needs an issue.

**Comment by alienzed** [comment]
Put


force_constant_lr = 2e-5

In your config prior to resuming.

**Comment by CognitiveDiffusion** [comment]
> Put
> 
> 
> force_constant_lr = 2e-5
> 
> In your config prior to resuming.

Thanks. I'll try that. Anywhere particular it should go?

**Comment by alienzed** [comment]
I have mine under warmup steps... no particular reason, but it works there.

**Comment by CognitiveDiffusion** [comment]
> I have mine under warmup steps... no particular reason, but it works there.

Hm... I tried using the forced 2e-5 lr now... and it still shows as lr=[0.0001] during training.

My last training run was cooked as well, so I think it actually did train at that too high lr.

Edit:

I even copied the last trainable Epoch to a new folder. Double checked the 2e-5 + forced lr ... still shows as 0.0001.

AdamW8BitKahan

Edit2:

Starting a new training run that initializes from the last stable Epoch worked and now shows the expected lr.

The other training run was at around 30k steps. Maybe that caused the issue somehow?

**Comment by alienzed** [comment]
Could be that it doesn't work for WAN...

Here's how I have it in my config, and it does work for me.
`
# Learning rate warmup.
warmup_steps = 100
force_constant_lr = 3e-5

eval_every_n_epochs = 1
eval_before_first_step = true
`

**Comment by CognitiveDiffusion** [comment]
> Could be that it doesn't work for WAN...
> 
> Here's how I have it in my config, and it does work for me.
> `
> # Learning rate warmup.
> warmup_steps = 100
> force_constant_lr = 3e-5
> 
> eval_every_n_epochs = 1
> eval_before_first_step = true
> `

Could be. I am training Wan T2V 14b and put it in my code the same way and it still went with 1e-4 during training.

I hope the initialized one works well.

**Comment by alienzed** [comment]
Maybe ask ChatGPT to look at the diffusion-pipe codebase and ask if it WAN training supports forcing the LR, that's how I've been working for a bit now. ChatGPT is far from always accurate, but with so few people being experts with this stuff, a powerful AI that can crawl the codebase is the next best thing.
just my two cents, good luck!

**Comment by CognitiveDiffusion** [comment]
> Maybe ask ChatGPT to look at the diffusion-pipe codebase and ask if it WAN training supports forcing the LR, that's how I've been working for a bit now. ChatGPT is far from always accurate, but with so few people being experts with this stuff, a powerful AI that can crawl the codebase is the next best thing.
> just my two cents, good luck!

I was kinda hoping Tdrussel would chime in.

I usually don't bug LLM's with tech that is so new.

**Comment by alienzed** [comment]
Sure, and apologies for not shutting up lol
the way I see it, tdrussel might not know, my understanding is that he is plugging other frameworks in, not writing them himself. This repo seems to be a bit of a beautiful frankenstein's monster integrating other code bases into one, but I could be wrong and maybe he wrote it all from the ground up, or maybe he does know the answer.
In any case, if ChatGPT gives you the right answer, you can simply share that with everyone here and let tdrussel focus on evolution vs tech support. That is what I did when I asked the same exact question but for HunyuanVideo. ChatGPT gave me the answer and it worked for me so I shared it here in a separate issue, it's now in the example if I am not mistaken!
ChatGPT doesn't need to know the answer, it can find it by crawling the code base :)
I'll step back now, hope I haven't been annoying here, good luck!

**Comment by tdrussell** [comment]
I tested Wan 14b training, with AdamW8bitKahan optimizer, putting ```force_constant_lr = 1e-5``` in the config when using ```--resume_from_checkpoint``` flag to resume training. It correctly sets the learning rate:

```
[2025-06-05 12:18:44,777] [INFO] [logging.py:107:log_dist] [Rank 0] step=12, skipped=0, lr=[1e-05], mom=[[0.9, 0.99]]
```
